### DISCUSSION TITLE
Differnt URLs for rewrites for different stages

### DISCUSSION THREAD

**Discussion by bastiW** [vercel/vercel#5076] [Help]
Hi,

I am currently adding the production stage to my enviroment. Therfore I need to change the rewrites depending on the stage. 

Currently I have:

```
{
  "alias": ["example.com"],
  "version": 2,
  "builds": [{ "src": "package.json", "use": "@now/static-build" }],
  "rewrites": [
    { "source": "/graphql/(.*)", "destination": "http://example-development-eb.eba.eu-central-1.elasticbeanstalk.com/graphql/$1" },
  ]
}
```

Now i would like to have something like that:
```
{
  "alias": ["example.com"],
  "version": 2,
  "builds": [{ "src": "package.json", "use": "@now/static-build" }],
  "rewrites": [
    { "source": "/graphql/(.*)", "destination": "@ELASTIC_BEANSTALK_URL/graphql/$1" }, //  HERE I WOULD LIKE TO USE THE VARIABLE
  ]
}
```

so ELASTIC_BEANSTALK_URL should come from a enviroment variable. 

How can I do that?

If not, what else can I do to set the rewrite URLs?

**Comment by styfle** [comment]
Hi @bastiW 

This is not possible at this time, you would need to make a Serverless Function to dynamically read the environment variable.

See a similar discussion #5062.

Also, you can remove `alias` and `builds` from your `vercel.json` since those are redundant.
